src/common/components/forms/CancelOfferForm.tsx

```diff
@@ -48,10 +52,31 @@ const CancelOfferForm: FC<CancelOfferFormProps> = ({
     if (!publicKey || !signTransaction || !offer || !nft) {
       return;
     }
+
+    const newActions: Action[] = [
+      {
+        name: `Canceling offer for ${Number(offer.price) / LAMPORTS_PER_SOL} SOL...`,
+        id: `cancelOffer`,
+        action: onCancelOffer,
+        param: undefined,
+      },
+    ];
+
     try {
-      await onCancelOffer();
-      toast.success(`Confirmed cancel offer success`);
-      await refetch();
+      await runActions(newActions, {
+        onActionSuccess: async () => {
+          toast.success(`Confirmed cancel offer success`);
+          await refetch();
+        },
+        onActionFailure: async (err) => {
+          toast.error(err.message);
```

Comment:
In other components you dont display the error inside `onActionFailure`, only in the outer `catch`. Is that intentional?

Reply:
the inner throws errors that are intended the external will catch unintended errors